03-devops-overview.Rmd

```diff
@@ -0,0 +1,183 @@
+4. Click "add respository" and give it a suitable name.
+
+This will create you a Git repo on DevOps. By default it will give you a main branch and a README file.
+
+### Clone to your computer
+
+Cloning a repository means that you:
+
+1. Get a copy of it on your computer.
+2. Your computer stores where to find the online copy so it can talk to the online one when you ask it to.
+
+To clone the blank repository to your computer you need to get the repo's URL.
+It is easiest to do this by:
+
+1. Going to "Files" in the "Repos" section of DevOps.
+2. Ensuring you are in the right repository by checking the name in the banner at the top of the screen.
+3. Clicking the clone button in the top right-hand corner.
+4. Copying the URL.
+
+Once you have copied the URL, you need to open git bash on your computer and clone the repository using the URL.
+When you open git bash you should do so in the folder where you want the repo to be.
+Ideally this will be on your C drive, outside of OneDrive or Teams, because git repositories don't work well if OneDrive
+or Teams are syncing them.
+There is no need for the additional syncing from OneDrive or Teams because DevOps provides the online back-up that others can access.
+*As long as you push your work regularly and at least once a day*, DevOps provides the back-up and any general rules than work should be on OneDrive or Teams to be backed up do not apply.
+
+One you've got git bash open, the clone command will look as follows:
+
+```sh
+git clone git_repo_url
+```
+
+If you are working on EDAP, it is best to clone the repo to your P drive.
+
+
+## Creating a git repository for a project you've been doing on your computer
+
+If you have already been working on a project locally you may wish to push this up and use this as the starting point for the collaborative workflow.
+
+In this case, create a new repository on DevOps but before you click the "Create" button after giving you repository a name, untick the "Add a README" box. This will give a truly blank repository.
+
+Once you click "Create" you will be take to a screen which says your new repository is empty.
+There will be an option to *Push an existing repository from command line*.
+
+In order to Push an existing repository you need to do the following:
+
+1. Open Git Bash in your local Git repo
+2. Copy the code from the DevOps section and run it
+
+This will set up the connection to DevOps and push an entire copy of the repo up to DevOps.
+
+## Adding users to a project
+
+DevOps is a restriced access code environment where users need to first be given permission to access the site as a whole and then individual projects. 
+
+To add users to a project:
+
+1. In the Overview section, go to the summary page 
+2. Click on the invite button in the top, right hand corner
+2. Search for user
+3. Click Add
+
+The user will then get a welcome email inviting them to the project.
+
```

Comment:
Maybe add something about levels of access? Stakeholder vs. basic. This has caused some problems in my team because someone only had stakeholder access and was unable to clone a repo, took us a while to figure out what was going on.